src/main/scala/org/apache/spark/sql/delta/DeltaOptions.scala

```diff
@@ -76,7 +76,7 @@ trait DeltaWriteOptionsImpl extends DeltaOptionParser {
    * MODIFY permissions, when schema changes require OWN permissions.
    */
   def canOverwriteSchema: Boolean = {
-    options.get(OVERWRITE_SCHEMA_OPTION).map(toBoolean(_, OVERWRITE_SCHEMA_OPTION)).getOrElse(false)
+    options.get(OVERWRITE_SCHEMA_OPTION).exists(toBoolean(_, OVERWRITE_SCHEMA_OPTION))
```

Comment:
This should be uniform throught this file, for example https://github.com/delta-io/delta/blob/master/src/main/scala/org/apache/spark/sql/delta/DeltaOptions.scala#L70 can also be converted to this

Reply:
@rahulsmahadev In the line 70, we can't use ```exists``` since the getOrElse doesn't take a false, but a default value.
Actually, ```exists``` is only equivalent to : 
```
.map(x => foo(x)).getOrElse(false)
```

Reply:
Oops my bad

Reply:
Will using exists over `.map(x => foo(x)).getOrElse(false)` have difference in performance ? Because I am inclined to keep the code uniform for the sake of readability. What do you think.

Reply:
That's Okay, Thanks for your feedback 🙏 

It think it's a small enhancement in the code performance. Let me explain : 
If we combine ```.map(x => foo(x))``` with ```.getOrElse(false)```, we will be calling this code in order to get the result :
```
if (isEmpty) false else { if (isEmpty) None else Some(foo(this.get))} .get
```

Explanation : here's the definition of ```map``` and ```getOrElse``` in the ```Options.scala``` class : 

```
@inline final def map[B](f: A => B): Option[B] =
    if (isEmpty) None else Some(f(this.get))
```

```
@inline final def getOrElse[B >: A](default: => B): B =
    if (isEmpty) default else this.get
```

But with ```exists```, we only do the following : 
```
!isEmpty && foo(this.get)
```

I believe that this small enhancement won't impact the code readability.